mcfix_base/models/ir_default.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2018 Creu Blanca
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+import json
+from odoo import api, models, tools
+
+
+class IrDefault(models.Model):
+    _inherit = 'ir.default'
+
+    @api.model
+    @tools.ormcache('self.env.uid', 'model_name', 'condition')
+    def get_model_defaults(self, model_name, condition=False):
+        """Related to  https://github.com/odoo/odoo/pull/23695 """
+        super(IrDefault, self).get_model_defaults(
+            model_name=model_name, condition=condition)
+        cr = self.env.cr
+        query = """ SELECT f.name, d.json_value FROM ir_default d
+                    JOIN ir_model_fields f ON d.field_id=f.id
+                    JOIN res_users u ON u.id=%s
+                    WHERE f.model=%s
+                        AND (d.user_id IS NULL OR d.user_id=u.id)
+                        AND (d.company_id IS NULL OR d.company_id={id1})
+                        AND {id2}
+                    ORDER BY d.user_id, d.company_id, d.id
+                """
+        company = self._context.get('default_company_id') or "u.company_id"
```

Comment:
tiny, but I prefer the variable to be company_id if it truly is (as in this case) an id, and not a company record